src/main/java/seedu/address/logic/ParserManager.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.logic;
+
+import seedu.address.logic.parser.FeatureParser;
+import seedu.address.logic.parser.ModuleListParser;
+import seedu.address.logic.parser.TodoListParser;
+import seedu.address.logic.parser.contactlistparsers.ContactListParser;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Represents the manager in charge of overseeing all the medium feature parsers.
+ */
+public class ParserManager {
+    private final ModuleListParser moduleListParser;
+    private final ContactListParser contactListParser;
+    private final TodoListParser todoListParser;
+
+    /**
+     * Creates a Container the holds all the medium parsers.
+     * @param moduleListParser parses module related commands.
+     * @param todoListParser parses task related commands.
+     * @param contactListParser parses contact related commands.
+     */
+    public ParserManager(ModuleListParser moduleListParser,
```

Comment:
Relating to the comment above, we could have ParserManager instantiate an instance for each of the 3 respective Parsers in the constructor here, rather than leave it to LogicManager since ParserManager is the one responsible for the 3 Parser classes. This is something we could consider.

Reply:
If we leave the instantiating of the 3 parsers in the constructor, then each time a command called by the user, we are re-creating a parser each time. And not reusing them, which i think is wrong.

Reply:
Yes, that sounds reasonable, ultimately ParserManager should be the one handling the instantiation of the related Parser classes.